app/models/directory_item.rb

```diff
@@ -53,38 +53,45 @@ def self.refresh_period!(period_type, force: false)
 
     ActiveRecord::Base.transaction do
       # Delete records that belonged to users who have been deleted
-      DB.exec "DELETE FROM directory_items
+      DB.exec("DELETE FROM directory_items
                 USING directory_items di
                 LEFT JOIN users u ON (u.id = user_id AND u.active AND u.silenced_till IS NULL AND u.id > 0)
                 WHERE di.id = directory_items.id AND
                       u.id IS NULL AND
-                      di.period_type = :period_type", period_type: period_types[period_type]
+                      di.period_type = :period_type", period_type: period_types[period_type])
 
       # Create new records for users who don't have one yet
-      DB.exec "INSERT INTO directory_items(period_type, user_id, likes_received, likes_given, topics_entered, days_visited, posts_read, topic_count, post_count)
+
+      column_names = DirectoryColumn.automatic_column_names + DirectoryColumn.plugin_directory_columns
```

Comment:
Is there any basic sanitizing we can do here on the column_names? For example make sure they only include `[a-z]` and `_`? Just trying to avoid injection vectors.

Reply:
Should this happen in the plugin api's `add_directory_column` and then we can assume from it's safe after that?

Reply:
Added the validation in the plugin api